bitcoin/src/Bitcoin/Transaction/Partial.hs

```diff
@@ -360,7 +360,7 @@ onPrevTxOut net signer tx ix input prevTxData =
   where
     newSigs = HM.mapWithKey sigForInput sigKeys
     sigForInput thePubKey theSecKey =
-        encodeTxSig . makeSignature net tx ix theSigInput $
+        maybe (error "Signature Gen Failed") encodeTxSig . makeSignature net tx ix theSigInput $
```

Comment:
Hm. Calling `error` in the library code isn't great. Is this known to never happen?

Reply:
Yeah this is one of those consequences of changing the signing function to return a maybe. It looks like you're already aware of [this](https://github.com/haskell-bitcoin/bitcoin/pull/44/files#r1585246017). I'll respond in that thread.